static/js/settings_empty_feed_notice.js

```diff
@@ -0,0 +1,16 @@
+import $ from "jquery";
+
+export function empty_feed_notice(table, table_widget, message) {
+    const $table_search = table.closest(".settings-section").find(".search").val();
+
+    if (table_widget.get_current_list().length === 0 && $table_search !== "") {
+        $(`.${table.attr("id")}`).addClass("empty-table");
+        table.attr("data-empty", `No ${message} match your current filter.`);
+    } else if (table_widget.get_current_list().length === 0) {
+        $(`.${table.attr("id")}`).addClass("empty-table");
+        table.attr("data-empty", `No ${message} found.`);
+    } else {
+        $(`.${table.attr("id")}`).removeClass("empty-table");
+        table.attr("data-empty", `No ${message} found.`);
+    }
+}
```

Comment:
Can't we just add this function in `list_widget.js`, and instead of calling it for each table from the respective settings file, we can add this functionality in the widget itself probably where `onUpdate` is called.

Also, please make sure that these messages are tagged for translation.

Reply:
If I understood you correctly, you would like me to add this function in list_widget.js and call it inside widget itself where ever needed, In this case inside onUpdate function. like this    ` ListWidget.empty_feed_notice();`